assets/src/edit-story/components/popup/utils.js

```diff
@@ -145,9 +145,12 @@ export function getOffset(placement, spacing, anchor, dock, popup) {
     bodyRect
   );
   const maxOffsetX = bodyRect.width - width - getXTransforms(placement) * width;
+
   // Vertical
   const offsetY = getYOffset(placement, spacingV, anchorRect);
-  const maxOffsetY = bodyRect.height + bodyRect.y - height;
+  const maxOffsetY =
+    bodyRect.height + bodyRect.y - height - getYTransforms(placement) * height;
```

Comment:
Small bug in the `Popup` utils, it wasn't taking placement into consideration when calculating the `maxOffsetY` (which it _was_ already doing for offsetX), this fixed my popup placement issues and all the other popups I tested continued to work as normal.